cpprevolve/revolve/gazebo/motors/PositionMotor.cpp

```diff
@@ -128,8 +129,20 @@ void PositionMotor::DoUpdate(const ::gazebo::common::Time &_simTime)
     position += (position > 0 ? -2 * M_PI : 2 * M_PI);
   }
 
+
   auto error = position - this->positionTarget_;
-  auto cmd = this->pid_.Update(error, stepTime);
+  auto cmd = this->pid_.Update(error, stepTime); // angular velocity TODO this is targeted velocity
+
+  if (this->battery_)
+  {
+      ::gazebo::physics::JointWrench jointWrench = this->joint_->GetForceTorque(0);
 
+      // TODO find which axis to use local or global
+      // TODO check the power if it should be positive or negative
+      // TODO change this for now im using the absolute value of the power so it always decreases from the joint movements
+      double power = -abs(cmd * jointWrench.body1Torque.Length()); // TODO check which torque to use 1 or 2
```

Comment:
floor when positive instead of abs

Reply:
CMD is the desired angular velocity (from PID) and not real velocity (from physics engine)

Reply:
Since the angular velocity can be both positive and negative, (leading to positive and negative power), therefore we need the absolute value and negate this to get the purely negative power usage.